### PR TITLE
Ants brain extraction interface

### PR DESCRIPTION
**Pull request by MartinPerez**
Here is a wrapper for the ants brain extraction script , #1230. Hope it will be useful to others as well. I provide test and also made a very small modification to antsCorticalThicknessoutputSpec not following the capital O in Output convention.

### CODE REVIEW

**Comment by chrisfilo** [comment]
please rebase/merge master - we have fixed the CircleCI tests

**Comment by MartinPerez** [comment]
Sorry for the delay. I rebased...

**Comment by MartinPerez** [comment]
tests still failing after rebase :S

File: nipype/interfaces/ants/segmentation.py [hunk 1]
```diff
@@ -594,6 +594,105 @@ def _list_outputs(self):
...
+    Examples
+    --------
+    >>> from nipype.interfaces.ants.segmentation import antsBrainExtraction
+    >>> brainextraction = antsBrainExtraction()
+    >>> brainextraction.inputs.dimension = 3
+    >>> brainextraction.inputs.anatomical_image ='T1.nii.gz'
+    >>> brainextraction.inputs.brain_template = 'study_template.nii.gz'
+    >>> brainextraction.inputs.brain_probability_mask ='ProbabilityMaskOfStudyTemplate.nii.gz'
+    >>> brainextraction.cmdline
+    'antsBrainExtraction.sh -a T1.nii.gz -m ProbabilityMaskOfStudyTemplate.nii.gz -e study_template.nii.gz -d 3 -s nii.gz -o highres001_
```
Comment (chrisfilo) [inline nipype/interfaces/ants/segmentation.py]: You are missing a quotation mark at the end. This makes one of the tests to fail: https://travis-ci.org/nipy/nipype/jobs/87317019#L4110

File: nipype/interfaces/ants/segmentation.py [hunk 2]
```diff
@@ -594,6 +594,105 @@ def _list_outputs(self):
...
+    >>> brainextraction.inputs.brain_template = 'study_template.nii.gz'
+    >>> brainextraction.inputs.brain_probability_mask ='ProbabilityMaskOfStudyTemplate.nii.gz'
+    >>> brainextraction.cmdline
+    'antsBrainExtraction.sh -a T1.nii.gz -m ProbabilityMaskOfStudyTemplate.nii.gz -e study_template.nii.gz -d 3 -s nii.gz -o highres001_'
+    """
+    input_spec = antsBrainExtractionInputSpec
+    output_spec = antsBrainExtractionOutputSpec
+    _cmd = 'antsBrainExtraction.sh'
+
+    def _format_arg(self, opt, spec, val):
```
Comment (blakedewey) [inline nipype/interfaces/ants/segmentation.py]: There shouldn't be any need to overload this function for these inputs as they are pretty straight forward. I would recommend removing this for code clarity.

Comment (MartinPerez) [inline nipype/interfaces/ants/segmentation.py]: you are right. got rid of the unnecessary overload!

**Comment by MartinPerez** [comment]
I think all is fine now.

File: nipype/interfaces/ants/segmentation.py [hunk 3]
```diff
@@ -594,6 +594,93 @@ def _list_outputs(self):
...
+        return super(ANTSCommand, self)._format_arg(opt, spec, val)
+
+    def _run_interface(self, runtime, correct_return_codes=[0]):
```
Comment (satra) [inline nipype/interfaces/ants/segmentation.py]: this function is also not necessary

File: nipype/interfaces/ants/segmentation.py (hunk 2 above)
Comment (satra) [inline nipype/interfaces/ants/segmentation.py]: still seeing the overloaded `_format_arg` .

**Comment by MartinPerez** [comment]
good to go?

File: nipype/interfaces/ants/segmentation.py [hunk 4]
```diff
@@ -594,6 +594,86 @@ def _list_outputs(self):
...
+                                              'To limit the metric computation to a specific region.'))
+    image_suffix = traits.Str('nii.gz', desc=('any of standard ITK formats,'
+                                              ' nii.gz is default'),
+                              argstr='-s %s', usedefault=True)
+    use_random_seeding = traits.Enum(0, 1, argstr='-u %d',
+                                     desc=('Use random number generated from system clock in Atropos'
+                                           '(default = 1)'))
+    keep_temporary_files = traits.Int(argstr='-k %d',
+                                      desc='Keep brain extraction/segmentation warps, etc (default = 0).')
+    use_floatingpoint_precision = traits.Enum(0, 1, argstr='-q %d',
```
Comment (satra) [inline nipype/interfaces/ants/segmentation.py]: @hjmjohnson - do all ants commands now have this option, if so should this be moved to the base class?

**Comment by hjmjohnson** [comment]
@satra all of ANTs does not have this command.  It is being implemented one program at a time.  :( Sorry.

**Comment by blakedewey** [comment]
@hjmjohnson is this true for the `--verbose` option? See my last commit in #1260.

**Comment by hjmjohnson** [comment]
This depends on what you consider an "ANTs" program.  There is no formal mechanism for tools build as part of the ANTs package to maintain a consistent interface.  There is an attempt to keep similar items consistent (via copy-n-paste coding) across tools, but it is not rigorously enforced.

I honestly do not know how uniform the --verbose option is across tools, that would require a review of the ANTs code base.

**Comment by blakedewey** [comment]
Should we merge this as is then?

**Pull request merged**